Fix mixed use of target_link_libraries() interfaces (plain/keyword)

--- CMakeLists.txt~	2023-12-13 03:08:50.000000000 +0100
+++ CMakeLists.txt	2024-01-09 15:04:18.628703991 +0100
@@ -115,7 +115,7 @@
   add_library(${PYWRAP} SHARED ${${PYWRAP}_SOURCES} ${${PYWRAP}_HEADERS})
   target_include_directories(${PYWRAP} SYSTEM PRIVATE ${PYTHON_INCLUDE_DIR})
-  target_link_libraries(${PYWRAP} ${PROJECT_NAME})
-  target_link_boost_python(${PYWRAP})
+  target_link_libraries(${PYWRAP} PUBLIC ${PROJECT_NAME})
+  target_link_boost_python(${PYWRAP} PUBLIC)
   get_relative_rpath(${${PYWRAP}_INSTALL_DIR} ${PYWRAP}_INSTALL_RPATH)
   set_target_properties(${PYWRAP} PROPERTIES INSTALL_RPATH
                                              "${${PYWRAP}_INSTALL_RPATH}")
